### PR TITLE
feat: implement notification image

### PR DESCRIPTION
**Pull request by sdowell**
Implement a notification image using the Argo notification-engine library. The notification controller is configured to watch either a RootSync or RepoSync in a single namespace.

### CODE REVIEW

File: cmd/notification/main.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		"Namespace of the Resource to be notified")
+	resyncCheckPeriod = flag.Duration("resync-period", pollingPeriod(util.NotificationResyncPeriod, time.Minute),
+		"Period of time between checking if the apiResource listener needs a resync")
+	cmName = flag.String("cm-name", os.Getenv(util.NotificationConfigMapName),
+		"Name of the ConfigMap for the notification configs")
+	secretName = flag.String("secret-name", os.Getenv(util.NotificationSecretName),
+		"Name of the Secret for the notification configs")
+)
+
+// pollingPeriod parses the polling duration from the environment variable.
```
Comment (nan-yu) [inline cmd/notification/main.go:74]: Duplicate with https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/util.go#L296.

Comment (sdowell) [reply cmd/notification/main.go]: Good catch, updated to use the shared util

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**